File: 2024_11_27/leetcode-228-mihai.py
from types import List

class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        ret = []
        rs = nums[0]
        re = nums[0]
        for x in nums[1:]:
            if x - re != 1:
                ret.append(f'{rs}->{re}' if rs != re else f'{rs}')
                rs = x
            re = x
        ret.append(f'{rs}->{re}' if rs != re else f'{rs}')

        return ret

File: 2024_11_27/leetcode-228-mihai.cs
public class Solution {
    public IList<string> SummaryRanges(int[] nums) {
        if(nums.Length == 0){
            return new List<string>();
        }
        var ret = new List<string>();
        int start = nums[0];
        int end   = nums[0];
        foreach(var n in nums){
            if(n == start){
                continue;
            }
            if(n - end != 1){
                if(start == end){
                    ret.Add(start.ToString());
                } else {
                    ret.Add(start.ToString() + "->" + end.ToString());
                }
                start = n;
            }
            end = n;
        }
        if(start == end){
            ret.Add(start.ToString());
        } else {
            ret.Add(start.ToString() + "->" + end.ToString());
        }
        return ret;
    }
}